### PR TITLE
[Merged by Bors] - chore(Data/Finset/Image): simplify proof of filter_mem_image_eq_image

### PR DESCRIPTION
**Pull request by b-mehta**
Simplify this proof - it's really just a combination of three existing facts.
The lemma `filter_mem_image_eq_image` is used nowhere in mathlib, so I wouldn't have an objection to deleting it instead...

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by YaelDillies** [comment]
This lemma should probably be killed altogether

File: Mathlib/Data/Finset/Image.lean [hunk 1]
```diff
@@ -439,6 +429,13 @@ theorem filter_image {p : β → Prop} [DecidablePred p] :
       ⟨by rintro ⟨⟨x, h1, rfl⟩, h2⟩; exact ⟨x, ⟨h1, h2⟩, rfl⟩,
        by rintro ⟨x, ⟨h1, h2⟩, rfl⟩; exact ⟨⟨x, h1, rfl⟩, h2⟩⟩
 
+theorem filter_mem_image_eq_image (h : ∀ x ∈ s, f x ∈ t) :
+    t.filter (· ∈ s.image f) = s.image f := by
+  rwa [filter_mem_eq_inter, inter_eq_right, image_subset_iff]
+
```
Comment (b-mehta) [inline Mathlib/Data/Finset/Image.lean:435]:
```suggestion
```

File: Mathlib/Data/Finset/Image.lean [hunk 2]
```diff
@@ -439,6 +429,9 @@ theorem filter_image {p : β → Prop} [DecidablePred p] :
       ⟨by rintro ⟨⟨x, h1, rfl⟩, h2⟩; exact ⟨x, ⟨h1, h2⟩, rfl⟩,
        by rintro ⟨x, ⟨h1, h2⟩, rfl⟩; exact ⟨⟨x, h1, rfl⟩, h2⟩⟩
 
+theorem fiber_nonempty_iff_mem_image (y : β) : (s.filter (f · = y)).Nonempty ↔ y ∈ s.image f := by
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Image.lean:432]: While we're at it, should `y` be implicit?
```suggestion
theorem fiber_nonempty_iff_mem_image {y : β} : (s.filter (f · = y)).Nonempty ↔ y ∈ s.image f := by
```

**Review by YaelDillies — COMMENTED** [review]
Can you update the PR title/description?

**Comment by Ruben-VandeVelde** [comment]
Let's deprecate the lemma first.

**Comment by b-mehta** [comment]
Yup, I agree that's a better idea.

**Comment by YaelDillies** [comment]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**